corehq/apps/enterprise/api/resources.py

```diff
@@ -138,4 +187,84 @@ def alter_list_data_to_serialize(self, request, data):
         return result
 
     def get_primary_key(self):
-        return 'domain'
+        return 'email'
+
+
+class MobileUserResource(ODataResource):
+    username = fields.CharField()
+    name = fields.CharField()
+    email = fields.CharField()
+    role = fields.CharField()
+    created_at = fields.DateTimeField()
+    last_sync = fields.DateTimeField()
+    last_submission = fields.DateTimeField()
+    commcare_version = fields.CharField(blank=True)
+    user_id = fields.CharField()
+    domain = fields.CharField()
+
+    def get_object_list(self, request):
+        account = BillingAccount.get_account_by_domain(request.domain)
+        report = EnterpriseMobileWorkerReport(account, request.couch_user)
+        return report.rows
+
+    def obj_get_list(self, bundle, **kwargs):
+        return self.get_object_list(bundle.request)
+
+    def dehydrate(self, bundle):
+        bundle.data['username'] = bundle.obj[0]
+        bundle.data['name'] = bundle.obj[1]
+        bundle.data['email'] = bundle.obj[2]
+        bundle.data['role'] = bundle.obj[3]
+        bundle.data['created_at'] = self.convert_datetime(bundle.obj[4])
+        bundle.data['last_sync'] = self.convert_datetime(bundle.obj[5])
+        bundle.data['last_submission'] = self.convert_datetime(bundle.obj[6])
+        bundle.data['commcare_version'] = bundle.obj[7]
+        bundle.data['user_id'] = bundle.obj[8]
+        bundle.data['domain'] = bundle.obj[10]
+
+        return bundle
+
+    def alter_list_data_to_serialize(self, request, data):
+        result = super().alter_list_data_to_serialize(request, data)
+        path = urljoin(request.get_full_path(), 'schema/#feed')
+        result['@odata.context'] = request.build_absolute_uri(path)
+
+        return result
+
+    def get_primary_key(self):
+        return 'user_id'
+
+
+class ODataFeedResource(ODataResource):
```

Comment:
Would you mind sticking in a docstring explaining this resource a bit?  domains and users are straightforward, but I don't think it's self-evident what this returns or how that would be used.  From context and prior experience I'm guessing it's a meta resource that makes the others discoverable?

Reply:
Added in a docstring, although I think it would helpful to know whether that aligns with your expectations. At least for me, this API is confusing because it intermingles summary information (feeds used, feeds available) with individual report data (the name of the report, how many rows are in that report). We're looking to move the summary information to the Project Spaces report, and simplify this to only list all OData Feeds within the Enterprise. Does this report become more intuitive with that change?

Reply:
That docstring is helpful - I was a bit lost when looking at this for the first time, felt like I had to dig to understand what the resource _was_ at a high level, and what you've put there now is sufficient to get that point across.